src/config.cpp

```diff
@@ -230,6 +230,8 @@ struct PipelineModel_Element : JSON::Element {
       v_.run_on_prompt = JSON::Get<bool>(value);
     } else if (name == "run_on_token_gen") {
       v_.run_on_token_gen = JSON::Get<bool>(value);
+    } else if (name == "reset_session_idx") {
```

Comment:
Is 'reset_session_idx' intuitive enough? I always get confused by the smart pointer reset/empty/clear naming. Would something like 'deallocate_session_index' be clear that the goal is to free memory?

Reply:
I would actually like to not have this exposed at all, and would like it to be a temporary solution (with the intention of deprecating the config in the future). But for now, some models are already using this config. So, probably not a good idea to change the name. I can add some comments in the config to explain what this option means.